test/manual/tls_support.test.ts

```diff
@@ -106,6 +108,44 @@ describe('TLS Support', function () {
       });
     });
   });
+
+  context('when tlsCertificateKeyFile is provided, but tlsCAFile is missing', () => {
+    let client: MongoClient;
+    beforeEach(() => {
+      client = new MongoClient(CONNECTION_STRING, {
+        tls: true,
+        tlsCertificateKeyFile: TLS_CERT_KEY_FILE,
+        serverSelectionTimeoutMS: 5000,
+        connectTimeoutMS: 5000
+      });
+    });
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    it('throws an error', async () => {
+      const err = await client.connect().catch(e => e);
+      expect(err).to.be.instanceOf(Error);
```

Comment:
I think one last thing here would be to check this is a `MongoServerSelectionError` instead of a generic `Error` (or at least something more specific)